server/game/gameSystems/DeployLeaderSystem.ts

```diff
@@ -17,7 +18,15 @@ export class DeployLeaderSystem<TContext extends AbilityContext = AbilityContext
     public eventHandler(event): void {
         Contract.assertTrue(event.card.isLeader());
 
+        const deployedEvent = new GameEvent(this.eventName, {
+            player: event.context.source.controller,
+            card: event.card,
+            context: event.context
+        });
+
+        event.context.game.addMessage('{0} plays {1}', event.context.player, event.context.source);
         event.card.deploy();
+        event.context.game.openEventWindow(deployedEvent, true);
```

Comment:
This is currently resolving the `deployedEvent` in a separate window than the `moveCard` action which happens inside of `deploy()`. It's effectively resolving them sequentially, but we need them to resolve in the same window simultaneously.

The way to do that is to have this event create a "contingent event", which means basically that when this event resolves it will create a second event that also resolves in the same window.

An example of how to set that up is in DefeatSystem below. You can see that the `updateEvent()` method calls a method in `CardTargetSystem`, which in turn calls `event.setContingentEventsGenerator`. You need to do the same but do it all inside of `DeployLeaderSystem.updateEvent()` and just have `event.setContingentEventsGenerator()` create the `deployedEvent`.

https://github.com/SWU-Karabast/forceteki/blob/1ce928369b827de16f5ff381ec531939f116c181/server/game/gameSystems/DefeatCardSystem.ts#L66

It's a little complex sorry, lmk if it would be helpful for me to provide more detail there. If it's helpful, you can see here where the event window generates the contingent events and adds them to the window to be resolved: https://github.com/SWU-Karabast/forceteki/blob/1ce928369b827de16f5ff381ec531939f116c181/server/game/core/event/EventWindow.js#L45